src/components/customers/EditCustomerDocumentLocaleDialog.tsx

```diff
@@ -0,0 +1,162 @@
+import { forwardRef } from 'react'
+import { gql } from '@apollo/client'
+import styled from 'styled-components'
+import { object, string } from 'yup'
+import { useFormik } from 'formik'
+
+import { Dialog, Button, DialogRef, Typography } from '~/components/designSystem'
+import { ComboBoxField } from '~/components/form'
+import { useInternationalization } from '~/hooks/core/useInternationalization'
+import {
+  EditCustomerDocumentLocaleFragment,
+  UpdateCustomerInput,
+  useUpdateCustomerDocumentLocaleMutation,
+} from '~/generated/graphql'
+import { theme } from '~/styles'
+import DocumentLocales from '~/public/documentLocales.json'
+import { addToast } from '~/core/apolloClient'
+
+gql`
+  fragment EditCustomerDocumentLocale on CustomerDetails {
+    id
+    name
+    externalId
+    billingConfiguration {
+      id
+      documentLocale
+    }
+  }
+
+  mutation updateCustomerDocumentLocale($input: UpdateCustomerInput!) {
+    updateCustomer(input: $input) {
+      id
+      billingConfiguration {
+        id
+        documentLocale
+      }
+    }
+  }
+`
+export interface EditCustomerDocumentLocaleDialogRef extends DialogRef {}
+
+interface EditCustomerDocumentLocaleDialogProps {
+  customer: EditCustomerDocumentLocaleFragment
+}
+
+const documentLocalesData: { value: string; label: string }[] = Object.keys(DocumentLocales).map(
+  (localeKey) => {
+    return {
+      value: localeKey,
+      // @ts-ignore
+      label: DocumentLocales[localeKey],
+    }
+  }
+)
+
+export const EditCustomerDocumentLocaleDialog = forwardRef<
+  DialogRef,
+  EditCustomerDocumentLocaleDialogProps
+>(({ customer }: EditCustomerDocumentLocaleDialogProps, ref) => {
+  const { translate } = useInternationalization()
+  const isEdition = !!customer.billingConfiguration?.documentLocale
+  const [updateDocumentLocale] = useUpdateCustomerDocumentLocaleMutation({
+    onCompleted(res) {
+      if (res.updateCustomer) {
+        addToast({
+          severity: 'success',
+          translateKey: isEdition
+            ? 'text_63ea0f84f400488553caa76f'
+            : 'text_63ea0f84f400488553caa77b',
+        })
+      }
+    },
+  })
+  const formikProps = useFormik<Pick<UpdateCustomerInput, 'billingConfiguration'>>({
+    initialValues: {
+      billingConfiguration: {
+        documentLocale: customer.billingConfiguration?.documentLocale,
+      },
+    },
+    validationSchema: object().shape({
+      billingConfiguration: object().shape({
+        documentLocale: string().required(''),
+      }),
+    }),
+    enableReinitialize: true,
+    validateOnMount: true,
+    onSubmit: async (values) => {
+      await updateDocumentLocale({
+        variables: {
+          input: {
+            id: customer.id,
+            ...values,
+            // TODO: Required by API, will fix those non-sense requirements later
```

Comment:
What's the non-sense here ?

Reply:
There is no sense to me to send the customer name and externalD in order to update a different attribute.

Only ID should be required imo.

Those are set as required today for some reason and I opened an improvement ticket to look into it and take more time on it later